### PR TITLE
doc/rados: edit stretch-mode procedure

### PR DESCRIPTION
**Pull request by zdover23**
Edit the "stretch mode" section in doc/rados/operations/stretch-mode.rst so that the procedure is formatted as a procedure and the sentences correctly have heads.

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "pacific"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

## Checklist
- Tracker (select at least one)
  - [ ] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [x] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [x] Updates relevant documentation
  - [ ] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
</details>

### CODE REVIEW

File: doc/rados/operations/stretch-mode.rst [hunk 1]
```diff
@@ -60,90 +60,102 @@ to avoid with standard CRUSH rules.
 
 Stretch Mode
 ============
-The new stretch mode is designed to handle the 2-site case. Three sites are
-just as susceptible to netsplit issues, but are much more tolerant of
-component availability outages than 2-site clusters are.
+Stretch mode is designed to handle situations in which you cannot guarantee the
```
Comment (anthonyeleven) [inline doc/rados/operations/stretch-mode.rst:63]: s/situations/topologies/

or maybe "deployments"

File: doc/rados/operations/stretch-mode.rst [hunk 2]
```diff
@@ -60,90 +60,102 @@ to avoid with standard CRUSH rules.
...
+when the cluster's CRUSH rule specifies that three copies are to be made, but 
+then a copy is placed in each data center with a ``min_size`` of 2. Under such
+conditions, a placement group can become active with two copies in the first
+data center and no copies in the second data center. 
+
+
+Entering Stretch Mode
+---------------------
 
 To enter stretch mode, you must set the location of each monitor, matching
```
Comment (anthonyeleven) [inline doc/rados/operations/stretch-mode.rst:74]: Enable instead of enter, perhaps?

File: doc/rados/operations/stretch-mode.rst [hunk 3]
```diff
@@ -60,90 +60,102 @@ to avoid with standard CRUSH rules.
...
+   monitor must be assigned a data center that is neither ``site1`` nor
+   ``site2``. For this purpose you can create another data-center bucket named
+   ``site3`` in your CRUSH and place ``mon.e`` there:
+
+   .. prompt:: bash $
+
+      ceph mon set_location e datacenter=site3
+      ceph mon enable_stretch_mode e stretch_rule datacenter
+
+When stretch mode is enabled, the OSDs will make PGs active only when they peer
```
Comment (anthonyeleven) [inline doc/rados/operations/stretch-mode.rst:131]: enabled, PGs will become `active` only when

File: doc/rados/operations/stretch-mode.rst [hunk 4]
```diff
@@ -60,90 +60,102 @@ to avoid with standard CRUSH rules.
...
+across data centers (or across whichever CRUSH bucket type was specified),
+assuming both are alive. Pools will increase in size from the default ``3`` to
+``4``, and two copies will be expected in each site. OSDs will be allowed to
+connect to monitors only if they are in the same data center as the monitors.
+New monitors will not be allowed to join the cluster if they do not specify a
+location.
+
+If all the OSDs and monitors from a data center become inaccessible at once,
```
Comment (anthonyeleven) [inline doc/rados/operations/stretch-mode.rst:139]: If all OSDs and monitors in one of the data centers become

**Review by anthonyeleven — APPROVED** [review]
suggestions

**Comment by zdover23** [comment]
> suggestions

I think we're going to use all of these suggestions. 

Getting stretch-mode.rst into shape is going to be a bit of a project. 

I'll integrate your changes into this PR after I've had a nap. I'm too tired right now to have good judgment.

**Comment by anthonyeleven** [comment]
> > suggestions
> 
> I think we're going to use all of these suggestions.
> 
> Getting stretch-mode.rst into shape is going to be a bit of a project.
> 
> I'll integrate your changes into this PR after I've had a nap. I'm too tired right now to have good judgment.

Groovy.  There are a lot of subtleties here in this fairly-new functionality; we must be cautious to not lead the reader stray.

**Comment by zdover23** [comment]
https://github.com/ceph/ceph/pull/51289 - Reef backport
https://github.com/ceph/ceph/pull/51290 - Quincy backport
https://github.com/ceph/ceph/pull/51291 - Pacific backport

**Pull request merged**